lib/delayed/worker.rb

```diff
@@ -142,17 +145,14 @@ def perform(job)
           job.destroy
         end
         job_say job, format('COMPLETED after %.4f seconds', run_time)
-        true # did work
-      rescue DeserializationError => e
-        job_say job, "FAILED permanently with #{e.class.name}: #{e.message}", 'error'
-
-        job.error = e
-        failed(job)
-        false # work failed
-      rescue Exception => e # rubocop:disable Lint/RescueException
-        self.class.lifecycle.run_callbacks(:error, self, job) { handle_failed_job(job, e) }
-        false # work failed
       end
+      true # did work
+    rescue DeserializationError => e
```

Comment:
Most of these `rescue`s were pulled one level _out_ of the corresponding lifecycle's `do` + `end`. This means that if something inside a lifecycle hook fails, the "after" hook for that lifecycle will not trigger, which I believe is more in line with expectations. (e.g. if `perform` fails, there is no `after(:perform)` callback fired, but there _is_ an `after(:error)` callback.)